nextcord/cache.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import annotations
+
+from typing import Dict, Iterable, Optional
+
+from .types.guild import Guild as GuildData
+from .types.message import Message as MessageData
+from .types.user import User as UserData
+
+__all__ = (
+    "BaseCache",
+    "MemoryCache",
+    "NullCache",
+)
+
+
+class BaseCache:
```

Comment:
Consider making this inherit `typing.Protocol` as well to help with tooling stuff

Reply:
Could also just be renamed to just `Cache`, the subclasses are alredy named specific enough imo